sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -42,16 +42,14 @@ object CommandUtils extends Logging {
 
   /** Change statistics after changing data by commands. */
   def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
-    if (table.stats.nonEmpty) {
```

Comment:
Because of this condition check, the insert table command will never able to calculate the table size even if user enables sparkSession.sessionState.conf.autoSizeUpdateEnabled.

Reply:
Thank you for pinging me, @sujith71955 . Got it. I'll take a look. BTW, I update the JIRA ID in the title.

Reply:
Sure . thanks